app/models/patient_session.rb

```diff
@@ -144,6 +144,11 @@ def pending_transfer?
     proposed_session_id.present?
   end
 
+  def remove_unrecorded_records
+    VaccinationRecord.where(patient_session: self).draft.destroy_all
+    GillickAssessment.where(patient_session: self).draft.destroy_all
```

Comment:
```suggestion
    draft_vaccination_records.destroy_all
    draft_gillick_assessments.destroy_all
```

This could be a nicer alternative, the relationships are defined here: https://github.com/nhsuk/manage-vaccinations-in-schools/blob/23ffaf1d94a04e43725d721176f4631a0d629623/app/models/patient_session.rb#L38-L54